src/owl/linalg/owl_linalg_intf.ml

```diff
@@ -3,18 +3,15 @@
  * Copyright (c) 2016-2019 Liang Wang <liang.wang@cl.cam.ac.uk>
  *)
 
-module type Base = sig
-  type elt
-  type mat
-  type complex_mat
-  type int32_mat
+module type Common = sig
+  include Owl_base_linalg_intf.Common
+  (* NOTE: functions below have not been implemented in Base Linalg *)
 
   (** {6 Basic functions} *)
 
   val inv : mat -> mat
```

Comment:
So here we have `val inv` in Common, and in L86 we have `include Owl_base_linalg_intf.Real`, which also includes an `inv` in Real. I'm not sure if that's intended.

Reply:
yea...so this is due to the fact that `inv` is only implemented for real matrices in base at the moment, but in core it's implemented for both real and complex matrices. Luckily duplicate `val` signatures doesn't seem to be a problem in ocaml...

Another option would be to move the signature of `inv` in base from `Real` to another module type called `To_be_implemented`. This module type will not be used in the core signatures. What do you think?

Reply:
If `inv` is the only special case here, I think it's OK leave it here for now. Could you please add some comment here such a TODO to note this issue, and I'll try to update the `inv` function to cover complex input soon.

Reply:
done!